### PR TITLE
fix: profile banner delay for orgs + small DX chore

### PR DESCRIPTION
**Pull request by leog**
## What does this PR do?

When creating an org, the owner should see a profile banner to set up their profile once they land on the app. That was delayed since we didn't trigger rendering the corresponding component once things change.

image

Also, when I was creating an org to replicate the issue and see the solution, noticed we sent the email with the code for lower envs either way and it was not needed since we introduced random totp codes for lower envs to stop relying on unstable dev mail services.

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)

## How should this be tested?

Have orgs enabled locally, create one going to '/settings/organizations/new` and after org wizard, see the banner appear when you land in the app.

## Mandatory Tasks

- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

### CODE REVIEW

File: apps/web/pages/event-types/index.tsx [hunk 1]
```diff
@@ -929,7 +929,7 @@ const EventTypesPage = () => {
       !!orgBranding && !document.cookie.includes("calcom-profile-banner=1") && !user?.completedOnboarding
     );
     // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, []);
+  }, [orgBranding, user]);
```
Comment (leog) [inline apps/web/pages/event-types/index.tsx:932]: This is the missing ingredient to fix this

File: packages/features/shell/Shell.tsx [hunk 2]
```diff
@@ -132,7 +132,6 @@ function useRedirectToLoginIfUnauthenticated(isPublic = false) {
 }
 
 function AppTop({ setBannersHeight }: { setBannersHeight: Dispatch<SetStateAction<number>> }) {
-  const router = useRouter();
```
Comment (leog) [inline packages/features/shell/Shell.tsx:135]: Unneeded code and also deprecated now

File: packages/trpc/server/routers/viewer/organizations/create.handler.ts [hunk 3]
```diff
@@ -131,6 +136,7 @@ export const createHandler = async ({ input, ctx }: CreateOptions) => {
 
     return { user: { ...createOwnerOrg, password } };
   } else {
+    if (!IS_PRODUCTION) return { checked: true };
```
Comment (leog) [inline packages/trpc/server/routers/viewer/organizations/create.handler.ts:139]: DX improvement to not send totp code email when creating an org on lower envs

File: apps/web/pages/event-types/index.tsx [hunk 4]
```diff
@@ -929,7 +929,7 @@
       !!orgBranding && !document.cookie.includes("calcom-profile-banner=1") && !user?.completedOnboarding
     );
```
Comment (sean-brydon) [inline apps/web/pages/event-types/index.tsx:930]: <i class='graphite__hidden'>[Re: lines 928 to 930]<br/></i>

NIT - this entire set should be in a separate useEffect as it has different dependancies  compared to the show intercom bubble.

<span class='graphite__hidden'><br/><br/>See this comment inline on <a href="https://app.graphite.dev/github/pr/calcom/cal.com/10546?utm_source=unchanged-line-comment">Graphite</a>.</span>

Comment (leog) [reply apps/web/pages/event-types/index.tsx:934]: Good catch, done

**Review by sean-brydon — APPROVED** [review]
Thanks for quick fix <3

**Pull request merged**